internal/services/securitycenter/security_center_subscription_pricing_resource.go

```diff
@@ -105,12 +127,20 @@ func resourceSecurityCenterSubscriptionPricingUpdate(d *pluginsdk.ResourceData,
 		}
 
 		if existing.Model != nil && existing.Model.Properties != nil && existing.Model.Properties.PricingTier != pricings_v2023_01_01.PricingTierFree {
-			return fmt.Errorf("the princing tier of this subscription is not Free \r %+v", tf.ImportAsExistsError("azurerm_security_center_subscription_pricing", id.ID()))
+			return fmt.Errorf("the pricing tier of this subscription is not Free \r %+v", tf.ImportAsExistsError("azurerm_security_center_subscription_pricing", id.ID()))
 		}
 	}
 
-	if v, ok := d.GetOk("subplan"); ok {
-		pricing.Properties.SubPlan = utils.String(v.(string))
+	if vSub, okSub := d.GetOk("subplan"); okSub {
+		pricing.Properties.SubPlan = utils.String(vSub.(string))
+	}
+
+	// can not set extensions for free tier
+	if pricing.Properties.PricingTier == pricings_v2023_01_01.PricingTierStandard {
+		if d.HasChange("extension") {
+			var extensions = expandSecurityCenterSubscriptionPricingExtensions(d.Get("extension").(*pluginsdk.Set).List())
```

Comment:
can we add a test to cover this scenario?

Reply:
I have done some manual tests since I am not able to run the tests through the framework it self, we can skip it for now but it was tested